cf-java-logging-support-servlet/src/main/java/com/sap/hcp/cf/logging/servlet/filter/CorrelationIdFilter.java

```diff
@@ -43,7 +52,10 @@ protected void beforeFilter(HttpServletRequest request, HttpServletResponse resp
 
     private String determineCorrelationId(HttpServletRequest request) {
         String correlationId = HttpHeaderUtilities.getHeaderValue(request, correlationHeader);
-        if (correlationId == null || correlationId.isEmpty() || correlationId.equals(Defaults.UNKNOWN)) {
+        if (isBlankOrDefault(correlationId)) {
+            correlationId = getCorrelationIdFromTraceparent(request);
```

Comment:
I guess this works, because correlationId is nullable, so the parsing can fail, and still work as expected.